### DISCUSSION TITLE
Some parts in the help are outdated

### DISCUSSION THREAD

**Discussion by tozGitHub** [WinEdt-Team/WinEdt-Support#29] [1.1 General Help and Discussion]
Dear Alex, dear community,

I found some parts where the help is outdated:

- WinEdt Manual|Dialogs...|Registration Data: the image refers to WinEdt 10.
- WinEdt Manual|Dialogs...|Dictionary Manager still shows the old picture.
- WinEdt Manual|How to: Examples|Install a Dictionary is still written for the old version where it was not possible to download dictionaries directly from the manager.

WinEdt version:  64-bit and Build: 20240319  (v. 1.2) - 64-bit (portable)
Windows version: 10.0.22631 Build 22631.

Best,
Thomas

**Comment by Alex-Simonic** [comment]
I updated a few things in the help and changed a few images but this is as far as I will go with the documentation updates at this stage.

Karl Koeller (Luca) was keeping WinEdt documentation and images up-to-date (his involvement is greatly missed). About six years ago he updated all help images (more than 150 of them) for WinEdt 10 with 96dpi screenshots on Windows 10. But not everything is 100% accurate anymore to reflect the changes in WinEdt 11.

I am working on 288dpi and wearing 3.5 X mag prescription glasses. Even if I were willing and able to update all images (which I am not) it would make the help file very large with high-definition screenshots. And for users working on lower dpi displays, it would probably look worse than now. Just the 12 updated images at 288dpi increased the size of the help file by more than .5MB... For now, this will have to do (unless somebody is willing and able to help with it). Years ago I had the complete WinEdt help sources and images (including a custom-written Delphi processor) available to the general public but nobody other than Karl was interested...

**Reply by Alex-Simonic** [reply]
While documenting the changes in Dictionary Manager I discovered and fixed a bug that could corrupt Dictionaries.ini when more than one mode filter was modified in the Dialog during the same session. It was introduced in WinEdt 11.0 (but never discovered or reported).

And the last-minute change: while documenting the new Zoom Panel functionality in the status line (with updated Status Line screenshots) I decided that the Console and other forms on the bottom of the screen should not be subject to font zooming. The defaults now use `FONT_MODE=3` instead of `FONT_MODE=1` for these forms (with Consolas 10 as the default font) in the `[FORMS]` configuration section. This will also work better for those who use large variable-width fonts for their documents...

These changes will be a part of the next beta build.

**Comment by urmel1960** [comment]
> Years ago I had the complete WinEdt help sources and images (including a custom-written Delphi processor) available to the general public but nobody other than Karl was interested...

I would be interested too.

**Reply by Alex-Simonic** [reply]
Great! I will write short instructions and make a zipped archive with all the documentation sources available.

This would be a good opportunity to create a new GitHub repository WinEdt-Team/WinEdt-Doc for WinEdt documentation. It would be great if someone with more GitHub experience joined WinEdt-Team on GitHub and helped set things up. This way other users can get involved with creating images, running text by the Language Tool or Grammarly, and more. The pre-processor written in Delphi (about 4.5K lines included with sources) currently creates output for MS HTML Help Compiler but (with some work) it can be modified to convert and extract it into TeX sources that can be compiled to pdf. I would help with this if there is any interest.

I don't have enough GitHub experience to start a project like this but I am willing to learn if somebody would help and provide guidance. An experiment like this would be a good test for the next stage when WinEdt sources will also be shared like this. Right now I don't see much point in dumping 200+K lines of Delphi source code for WinEdt there but the day will come...

Any thoughts on this?

**Reply by urmel1960** [reply]
Thanks Alex, as soon as you're ready I'll download the archive and take a look at it. 
We use a [commercial tool ](https://www.helpscribble.com/index.html) for creating help files for our projects here. It may also be useful in this case. I will have a look. 

I have 144 dpi on my computer - let's see if my screenshots will be usable, what size they are and whether it might be possible to work on the required resolutions automatically or reduce the image sizes.

I'm not the GitHub guru - I use a self-hosted GitLab EE here. But I'm happy to help if I can.

**Reply by Alex-Simonic** [reply]
Download:

https://www.winedt.com/work/WinEdtHelp.zip

Unzip the Help folder somewhere where you are allowed to write.

Carefully read the file "Help\Doc\0Readme.txt" for instructions.

**Reply by tozGitHub** [reply]
Dear Alex, dear Urmel, dear community,

I got inspired by your answer and thought about how to make the process of generating the images as automatic as possible. I came up with some ideas and sketches of programs. It is mainly for taking the pictures of the dialog boxes. However, my conclusion at the end is that maybe it would be best to include the ability of making screenshots directly into WinEdt (but maybe use some of my ideas nonetheless as inspiration).
# Strategy

The strategy is as follows:
1. Execute a WinEdt Macro that initiates taking all the screenshots
2. For each screenshoot in question
 2.1. Start a new WinEdt instance
 2.2. Open the dialog
 2.3. Execute a Python program taking the screenshot
 2.4. Close the WinEdt instance

During the process, I figured out that the point about opening the dialog should be enhanced. Instead of executing the dialog, we also should be able to execute macros to prepare WinEdt for the correct state before opening the dialog.
# Parts of the code
## Screenshots.edt
The file is basically a sequence of commands taking screenshots, except the first line is for setting the path and has to be adjusted.
```
LetReg(-8,"C:\Users\Thomas\AppData\Roaming\WinEdt\WinEdtHelp\Test");
Exe('%-8\TakeScreenShot.edt','',1,"WinEdt Configuration Wizard","WinEdtWizard('WinEdt Configuration Wizard');","Wizard");
Exe('%-8\TakeScreenShot.edt','',1,"License Monitor","ShowLicenseMonitor;","License Monitor");
Exe('%-8\TakeScreenShot.edt','',1,"MUI","!MuiTPanel","TPanel");
```
The varying arguments are as follows (let us look at the last example):
- MUI: name of the window in order that the program takes the screenshot of the right dialog box.
- !MuiPanel: command to execute; when there is an exclamation mark in front, then this is the name of the macro to execute.
- TPanel: name of the file to which to save the screenshoot.
## TakeScreenShot.edt
```
// 0: Name of the window
// 1: Command to execute
// 2: Filename
Wait(9999);
Wait(9999);
Wait(9999);
// Has to be adjusted
LetReg(-9,"C:\Users\Thomas\AppData\Local\Programs\Python\Python310\python.exe");
LetReg(-8,"C:\Users\Thomas\AppData\Roaming\WinEdt\WinEdtHelp\Test");
ExtractByIndex(-7, "%!1", 0, 0);
IfStr("%-7","!","=","JMP('Macro');","JMP('Execute');");
:Macro::
GetLength(-6, "%!1");
ExtractByIndex(1,"%!1",1,%-6);
WinExe("", |%B\WinEdt.exe -C='Hi' "[Exe('%-8\%!1.edt');]"|);
JMP("Resume");
:Execute::
WinExe("", |%B\WinEdt.exe -C='Hi' "[%!1]"|);
:Resume::
Wait(9999);
// 0: Name of the script
// 1: Path
// 2: Name of the Window
// 3: Filename
WinExe("", "%-9 %-8\TakeScreenShoot.py %-8 ""%!0"" ""%!2""" , "%-8", "", 1000,0, "", "", "",11);
//WinExe("", "%-9 %-8\TakeScreenShoot.py %-8 ""%!0"" ""%!2""" , "%-8", "", 0,0, "", "", "",11);
Wait(9999);
WinExe("", |%B\WinEdt.exe -C='Hi' "[ExitWinEdt;]"|);  
```
Not that much is going on. The first part are just setting some paths. After this, there is just some amount of code to figure out whether we want to execute a macro or not and then takes care of it. Then the WinEdt instance is started. After this, we execute the Python script that takes the screenshot.
At the end, WinEdt is closed again.
### Issues with the code
There are more or less unmotivated ```Wait```'s all over the place. Since it is only a proof of concept, I did not go for a beauty contest.
## TakeScreenShoot.py
```Python
# -*- coding: utf-8 -*-
"""
Created on Sat Mar 30 10:56:02 2024

@author: Thomas
"""

# https://www.youtube.com/watch?v=OlEEv7lLbW0
# (A fellow named Zubair explains how to take screenshots in Python)
# https://stackoverflow.com/questions/2090464/python-window-activation
# (luc, https://stackoverflow.com/users/117092/luc, answeres the question by
#  exupero, https://stackoverflow.com/users/158002/exupero)
# https://www.geeksforgeeks.org/command-line-arguments-in-python/

import pygetwindow
import pyautogui
from PIL import Image

import win32gui
import re
import sys


path=sys.argv[1]
windowName=sys.argv[2]
name=sys.argv[3]

class WindowMgr:
    """Encapsulates some calls to the winapi for window management"""
    
    def __init__ (self):
        """Constructor"""
        self._handle = None
    
    def find_window(self, class_name, window_name=None):
        """find a window by its class_name"""
        self._handle = win32gui.FindWindow(class_name, window_name)
    
    def _window_enum_callback(self, hwnd, wildcard):
        """Pass to win32gui.EnumWindows() to check all the opened windows"""
        if re.match(wildcard, str(win32gui.GetWindowText(hwnd))) is not None:
            self._handle = hwnd
    
    def find_window_wildcard(self, wildcard):
        """find a window whose title matches the wildcard regex"""
        self._handle = None
        win32gui.EnumWindows(self._window_enum_callback, wildcard)
    
    def set_foreground(self):
        """put the window in the foreground"""
        win32gui.SetForegroundWindow(self._handle)


def main():
  w = WindowMgr()
  windowNameZ=""
  # For different languages
  wName=windowName.split("|")
  for name in wName:
      try:
          w.find_window_wildcard(".*"+name+".*")
          w.set_foreground()
          windowNameZ=name
          print("OK: "+windowNameZ)
          break
      except:
          print("Failed: "+name)
          3*3
  for wiName in wName:
    try:
        window = pygetwindow.getWindowsWithTitle(wiName)[0]
        left, top = window.topleft
        right, bottom = window.bottomright
        pyautogui.screenshot(path+"\\"+name+".png")
        im=Image.open(path+"\\"+name+".png")
        im=im.crop((left+10, top, right-10, bottom-10))
        im.save(path+"\\"+name+".png")
        print("OK pyget")
        break
    except:
        3*3
        print("Failed: pyget")

if __name__ == "__main__":
    main()

```
## Spreadsheet for generating the commands
I think the easiest way to generate the commands for the individual screenshots is with the help of a spreadsheet program. This way, we can easily generate a column featuring all necessary commands. The concatenation could be done as follows (probably needs adjustment for the program in use):
```
=CONCAT("Exe('%-8\TakeScreenShot.edt','',1,'";C4;"','";B4;"','";D4;"');")
```
Here is what I had filled in the table when I decided that I should first find out whether continuing makes sense at all.

|                  |                                                                                                                                |                             |                           |                            |
| ---------------- | ------------------------------------------------------------------------------------------------------------------------------ | --------------------------- | ------------------------- | -------------------------- |
| Window Name      | Command                                                                                                                        | Window Name                 | Filename                  | Filename                   |
|                  |                                                                                                                                |                             |                           | auto-begin-matrix          |
|                  |                                                                                                                                |                             |                           | auto-begin                 |
|                  | Browse(0,"TeX\|*.tex\|All Files\|*.*","%B\Samples\Thesis");                                                                    | Open                        | Browse                    | Browse                     |
|                  | !FindInFiles                                                                                                                   | FindInFiles                 | Collect                   | Collect                    |
|                  | GetColor(0,"");                                                                                                                | Color                       | Colors Dialog             | Colors Dialog              |
|                  |                                                                                                                                | Command Prompt              | Command Prompt            | Command Prompt             |
|                  | CompareDocuments;                                                                                                              | Compare Documents           | Compare                   | Compare                    |
|                  |                                                                                                                                | Console                     | Console                   | Console                    |
|                  | !DelFiles                                                                                                                      | Erase Ouput Files           | DelFiles                  | DelFiles                   |
|                  |                                                                                                                                |                             |                           | DocTabs                    |
|                  | DocumentSettings(6);                                                                                                           | Document Settings           | Document CP Converter     | Document CP Converter      |
|                  | DocumentSettings(3);                                                                                                           | Document Settings           | Document Format           | Document Format            |
|                  |                                                                                                                                | Document List               | Document List             | Document List              |
|                  | DocumentSettings(2);                                                                                                           | Document Settings           | Document Options          | Document Options           |
|                  | DocumentSettings(1);                                                                                                           | Document Settings           | Document Settings         | Document Settings          |
|                  | DocumentSettings(4);                                                                                                           | Document Settings           | Document Status           | Document Status            |
|                  | DocumentSettings(5);                                                                                                           | Document Settings           | Document Time Stamps      | Document Time Stamps       |
|                  |                                                                                                                                | dvi2ps                      | dvips                     | dvips                      |
|                  | EnterLongReg(0,"Label","Caption");                                                                                             | Caption                     | EnterLongReg              | EnterLongReg               |
|                  | EnterReg(0,"Label","Caption");                                                                                                 | Caption                     | EnterReg                  | EnterReg                   |
|                  | EnterString("Enter SRC Switch","SRC Switch",0);                                                                                | SRC Switch                  | EnterString               | EnterString                |
|                  | EnumerateLines;                                                                                                                | Enumerate Lines             | Enumerate                 | Enumerate                  |
|                  |                                                                                                                                | PDFTeXify                   | Error                     | Error                      |
|                  |                                                                                                                                | Execution Modes             | Exec-AUX                  | Exec-AUX                   |
|                  |                                                                                                                                | Execution Modes             | Exec-GS                   | Exec-GS                    |
|                  |                                                                                                                                | Execution Modes             | Exec-PATH                 | Exec-PATH                  |
|                  |                                                                                                                                | Execution Modes             | Exec-PDF                  | Exec-PDF                   |
|                  |                                                                                                                                | Execution Modes             | Exec-TeX                  | Exec-TeX                   |
|                  |                                                                                                                                | Execution Modes             | Exec-Var                  | Exec-Var                   |
|                  |                                                                                                                                | Execution Modes             | Exec                      | Exec                       |
|                  | ExternalPrint(!\|Run('dvips.exe %? "%N"', "%P", 0,0, "dvips...", 0,1)\|,"#0 #3 #1 #2","Dvips...","-p","-l","","-t landscape"); | Dvips                       | ExternalPrint             | ExternalPrint              |
|                  |                                                                                                                                | File List                   | File List                 | FileList                   |
|                  |                                                                                                                                | WinEdt TeX Icons            | Filetypes                 | Filetypes                  |
|                  |                                                                                                                                |                             |                           | Find-popup                 |
| FindDialog(0);   |                                                                                                                                | Find Text                   | Find                      | Find                       |
|                  |                                                                                                                                | Find in Files               | FindinFiles               | FindFind                   |
|                  |                                                                                                                                | Find Text                   | Find-popup                | FindinFiles                |
|                  |                                                                                                                                | Gather                      | Gather                    | Gather                     |
|                  | GetCounter("Items:","Cases");                                                                                                  | Cases                       | GetCounter                | GetCounter                 |
|                  | GetDim;                                                                                                                        | Dimensions                  | GetDim                    | GetDim                     |
|                  | GetLongString("Prompt","Caption");                                                                                             | Caption                     | GetLongString             | GetLongString              |
|                  | GetString("Prompt","Caption");                                                                                                 | Caption                     | GetString                 | GetString                  |
|                  |                                                                                                                                |                             |                           | GUI                        |
|                  | GetImage("Bookmark");                                                                                                          | Images                      | Images                    | Images                     |
|                  | IncrementalSearch;                                                                                                             | Incremental Search          | Incremental               | Incremental                |
|                  |                                                                                                                                |                             |                           | Insert and Goto Line       |
|                  | InsertItem('%B\Config\Edt\Var.dat','','Insert %%Parameter',1,1);                                                               | Insert %Parameter           | InsertItem                | InsertItem                 |
|                  | ShowLicenseMonitor;                                                                                                            | License Monitor             | License Monitor           | License Monitor            |
|                  |                                                                                                                                | WinEdt Links                | Links                     | Links                      |
|                  |                                                                                                                                |                             |                           | mui                        |
|                  | NewDoc("",1);                                                                                                                  | New Document                | NewDoc                    | NewDoc                     |
|                  |                                                                                                                                | Options Interface           | Options-Popup             | Options-Popup              |
|                  |                                                                                                                                | Options Interface           | Options                   | Options                    |
|                  |                                                                                                                                |                             |                           | Popup                      |
|                  |                                                                                                                                |                             |                           | PopupControls              |
|                  |                                                                                                                                |                             |                           | PopupText                  |
|                  |                                                                                                                                | Preferences                 | Prefs-Appearance          | Prefs-Appearance           |
|                  |                                                                                                                                | Preferences                 | Prefs-Backup              | Prefs-Backup               |
|                  |                                                                                                                                | Preferences                 | Prefs-Editor              | Prefs-Editor               |
|                  |                                                                                                                                | Preferences                 | Prefs-Font                | Prefs-Font                 |
|                  |                                                                                                                                | Preferences                 | Prefs-Hightlight          | Prefs-Highlight            |
|                  |                                                                                                                                | Preferences                 | Prefs-Misc                | Prefs-Misc                 |
|                  |                                                                                                                                | Preferences                 | Prefs-Mouse               | Prefs-Mouse                |
|                  |                                                                                                                                | Preferences                 | Prefs-Status              | Prefs-Status               |
|                  |                                                                                                                                | Preferences                 | Prefs-Tabs                | Prefs-Tabs                 |
|                  |                                                                                                                                | Preferences                 | Prefs-Unicode             | Prefs-Unicode              |
|                  |                                                                                                                                | Preferences                 | Prefs                     | Prefs                      |
|                  | PrintDoc;                                                                                                                      | Print                       | Print                     | Print                      |
| ShowProjectList; |                                                                                                                                | Project List                | PrjList                   | PrjList                    |
|                  |                                                                                                                                |                             |                           | Progress                   |
| ProjectManager;  |                                                                                                                                | Project Manager             | Project Manager           | Project Manager            |
|                  |                                                                                                                                |                             |                           | Project Menu               |
|                  | Prompt("(x) Yes No",3,3,"","");                                                                                                | Error                       | Prompt                    | Prompt                     |
|                  | RepeatMacro;                                                                                                                   | Macro Recorder              | Record                    | Record                     |
|                  |                                                                                                                                | Macro Recorder              | Recorder                  | Recorder                   |
|                  |                                                                                                                                | Replace Text                | Replace                   | Replace                    |
|                  | RunPrompt("Run Macro ...",1);                                                                                                  | Run Macro                   | RunMacro                  | RunMacro                   |
|                  |                                                                                                                                |                             |                           | Set Main File              |
|                  | PrintDoc;                                                                                                                      | Print                       | Setup                     | Setup                      |
|                  | GetShortcut("Ctrl+F");                                                                                                         | Shortcuts                   | Shortcuts                 | Shortcuts                  |
| Sort;            |                                                                                                                                | Sort Lines                  | Sort                      | Sort                       |
|                  | SpellCheck;                                                                                                                    | Spell Checking              | Spell                     | Spell                      |
|                  |                                                                                                                                |                             |                           | Spelling                   |
|                  |                                                                                                                                |                             |                           | Suggestions                |
|                  |                                                                                                                                |                             |                           | Tree                       |
|                  |                                                                                                                                |                             |                           | TUG                        |
|                  |                                                                                                                                |                             |                           | UAC                        |
|                  | Variables;                                                                                                                     | String Variables            | Variables                 | Variables                  |
|                  |                                                                                                                                |                             |                           | WinEdt                     |
|                  | WinEdtWizard('WinEdt Configuration Wizard');                                                                                   | WinEdt Configuration Wizard | Wizard                    | Wizard                     |
|                  |                                                                                                                                | Word Count                  | WordCount                 | WordCount                  |
|                  |                                                                                                                                |                             |                           |                            |
|                  | AboutWinEdt;                                                                                                                   | About WinEdt                | 288dpi\About              | 288dpi\About               |
|                  |                                                                                                                                | Bibliography Items          | 288dpi\cite               | 288dpi\cite                |
|                  | DictionaryManager("User (Addon)");                                                                                             | Dictionary Manager          | 288dpi\Dictionary Manager | 288dpi\Dictionary Manager  |
|                  |                                                                                                                                |                             |                           | 288dpi\GUI                 |
|                  |                                                                                                                                |                             |                           | 288dpi\GUIDark             |
|                  | RegisterWinEdt;                                                                                                                | Registration Data           | 288dpi\register           | 288dpi\register            |
|                  |                                                                                                                                |                             |                           | 288dpi\Status Line Toggled |
|                  |                                                                                                                                |                             |                           | 288dpi\Status Line         |
|                  |                                                                                                                                |                             |                           | 288dpi\Status LineDark     |
|                  |                                                                                                                                |                             |                           | 288dpi\ToolbarDef          |
|                  |                                                                                                                                |                             |                           | 288dpi\ToolbarDefDark      |
|                  |                                                                                                                                |                             |                           | 288dpi\Toolbars            |
|                  |                                                                                                                                |                             |                           |                            |
|                  |                                                                                                                                |                             |                           | mui\bvLowered              |
|                  |                                                                                                                                |                             |                           | mui\bvNone                 |
|                  |                                                                                                                                |                             |                           | mui\bvRaised               |
|                  |                                                                                                                                |                             |                           | mui\csDropDown             |
|                  |                                                                                                                                |                             |                           | mui\csDropDownList         |
|                  |                                                                                                                                |                             |                           | mui\csSimple               |
|                  |                                                                                                                                |                             |                           | mui\psnup                  |
|                  |                                                                                                                                |                             |                           | mui\TButton                |
|                  |                                                                                                                                |                             |                           | mui\TCheckBox              |
|                  |                                                                                                                                |                             |                           | mui\TComboBox-Font         |
|                  |                                                                                                                                |                             |                           | mui\TComboBox-Simple       |
|                  |                                                                                                                                |                             |                           | mui\TEdit                  |
|                  |                                                                                                                                |                             |                           | mui\TGroupBox              |
|                  |                                                                                                                                |                             |                           | mui\TLabel                 |
|                  |                                                                                                                                |                             |                           | mui\TListBox               |
|                  |                                                                                                                                |                             |                           | mui\TPageControl           |
|                  |                                                                                                                                |                             |                           | mui\TPanel                 |
|                  |                                                                                                                                |                             |                           | mui\TRadioButton           |
|                  |                                                                                                                                |                             |                           | mui\TRadioGroup            |
|                  |                                                                                                                                |                             |                           | mui\TSpeedButton           |
|                  |                                                                                                                                |                             |                           | mui\TTabSheet              |
|                  |                                                                                                                                |                             |                           |                            |
|                  |                                                                                                                                |                             |                           | Colors and Selections      |

## DelFiles.edt
This is just one example of what I mean by executing a whole macro file.
It is for creating DelFiles.png:
```
Open("%b\Samples\Thesis\Thesis.tex");
ClearFileItems;
AddFileItem(0,"PDF File","%N.pdf","TeX");
AddFileItem(0,"SYNC File","%N%$('PDF-SyncEx');","TeX");

AddFileItem(1,"DVI File","%N.dvi","TeX");
AddFileItem(1,"PS  File","%N.ps","TeX");

AddFileItem(1,"LOG Files","*.log","");
AddFileItem(1,"AUX Files","*.aux","TeX");
AddFileItem(1,"BBL Files","*.bbl","TeX");
AddFileItem(1,"BLG Files","*.blg","TeX");
AddFileItem(1,"IDX Files","*.idx","TeX");
AddFileItem(1,"IND Files","*.ind","TeX");
AddFileItem(1,"ILG Files","*.ilg","TeX");
AddFileItem(1,"GLO Files","*.glo","TeX");
AddFileItem(1,"GLS Files","*.gls","TeX");
AddFileItem(1,"GLG Files","*.glg","TeX");
AddFileItem(1,"LOF Files","*.lof","TeX");
AddFileItem(1,"LOT Files","*.lot","TeX");
AddFileItem(1,"TOC Files","*.toc","TeX");
AddFileItem(1,"OUT Files","*.out","TeX");

AddFileItem(1,"MPX Files","*.mpx","MetaPost");

AddFileItem(1,"BAK Files","*.bak","");
AddFileItem(1,"SAV Files","*.sav","");
AddFileItem(1,"TMP Files","*.tmp","");

AddFileItem(1,"TEMP Files","_temp.*","TeX");

EraseWorkingFiles("%b\Samples\Thesis\", "Erase Output Files"); 
```

# Issues
- It seems that I can only catch windows that come from dialogs, and not for example the Options Interface. There might be different remedies:
  - Someone finds out how to do it in Python
  - Someone can do it with another program (Autohotkey might be a candidate)
  - Alex adds some macro to WinEdt such that we can get the coordinates of windows (or even the screenshots themselves)
- If more than one WinEdt instance is open, then the screenshot might be taken from the wrong istance.
- Taking the screenshot of the Command Prompt needs some commands to be executed there. Maybe the easiest way to get it is using a batch file.
- The command ShowExecutionModes might now allow for the selection of the panels.
- The elements of the MUI have to be created.
# Conclusion
Just Googling, it seems to be possible to take screenshots in Delphi as well. Maybe this could just be a new macro (or family of macros) that take screenshots of various elements included into WinEdt itself. In my opinion, this would be the best solution. This then also would enable everyone to generate the images by themselves and hence getting the same look as they have in their specific WinEdt.

Best,
Thomas

**Reply by Alex-Simonic** [reply]
Thanks for giving this some thought. While having the process automated has many advantages it may take too much effort to do so. The devil is in the details. For now, I will not be adding extra macros for WinEdt Dialogs and controls taking selfies. Maybe one day for the future versions...
